### PR TITLE
Create README for Section 3

### PR DESCRIPTION
**Pull request by kingthorin**
- Added README.md in side chapter 3.
- Trimmed colons from anchors in both the new README.md in chapter 3 and the base README. They don't seem to be accepted on either github or the www site.
  - Ex GitHub:
https://github.com/OWASP/wstg/blob/master/document/3-The_OWASP_Testing_Framework/0-The_OWASP_Testing_Framework.md#Phase-2:-During-Definition-and-Design
vs https://github.com/OWASP/wstg/blob/master/document/3-The_OWASP_Testing_Framework/0-The_OWASP_Testing_Framework.md#Phase-2-During-Definition-and-Design
  - Ex www: https://owasp.org/www-project-web-security-testing-guide/latest/3-The_OWASP_Testing_Framework/0-The_OWASP_Testing_Framework.html#phase-2:-during-definition-and-design
vs
https://owasp.org/www-project-web-security-testing-guide/latest/3-The_OWASP_Testing_Framework/0-The_OWASP_Testing_Framework.html#phase-2-during-definition-and-design

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
@rejah does trimming the colons in the anchor links present any problem for the PDF job?

File: document/3-The_OWASP_Testing_Framework/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# The OWASP Testing Framework
+
+[3.1 Overview](0-The_OWASP_Testing_Framework.md#Overview)
```
Comment (rejahrehim) [inline document/3-The_OWASP_Testing_Framework/README.md:3]: Can we change it to `3.1  [Overview](0-The_OWASP_Testing_Framework.md#Overview)`

File: document/3-The_OWASP_Testing_Framework/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+[3.2 Phase 1: Before Development Begins](0-The_OWASP_Testing_Framework.md#Phase-1-Before-Development-Begins)
```
Comment (rejahrehim) [inline document/3-The_OWASP_Testing_Framework/README.md:5]: Same as above

File: document/3-The_OWASP_Testing_Framework/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+[3.3 Phase 2: During Definition and Design](0-The_OWASP_Testing_Framework.md#Phase-2-During-Definition-and-Design)
```
Comment (rejahrehim) [inline document/3-The_OWASP_Testing_Framework/README.md:7]: Same as above. For all the following links too

**Comment by rejahrehim** [comment]
> @rejah does trimming the colons in the anchor links present any problem for the PDF job?

No. That is good

**Pull request merged**